repeat/p5js/hidden-circles/index.js

```diff
@@ -0,0 +1,69 @@
+var dis = 20;
+
+function setup() {
+  createCanvas(500, 500);
+  background(255);
+}
+
+function drawLines(count, distance) {
+  distance = distance || 20;
+  
+  for (var i = 0; i < count; i++) {
+     var x = i * distance;
+    
+    line(x, 0, x, 500);
+  }
+}
+
+function draw() {
+  stroke(0);
+  strokeWeight(2);
+  drawLines(100, dis);
+  
+  strokeWeight(10);
+  stroke(255, 255, 255);
```

Comment:
Most of these magic numbers are kind of enforced by the poor design of the API of the p5 framework. Take for example the stroke function. What do the parameters do? Seems like they are the `r`, `g`, and `b` part of a color. But can you be sure, that it is not `b`, `r`, and `g`? Normally you would counter these "magic" numbers with variables. But what would that mean?

```jsx
const strokeWeightValue = 10 // in px and value so that it does not clash with the magic global function

strokeWeight(strokeWeightValue) // not more descriptive

const strokeRedPart = 255
const strokeBluePart = 255
const strokeGreenPart = 255

stroke(strokeRedPart, strokeGreenPart, strokeBluePart) // wtf
```

A better API would make these "magic" numbers more expressive by design.

```jsx
strokeWidth({ px: 10 })

stroke({ r: 255, g: 255, b: 255 })
```

This is true for basically every function that needs to be used. The way the API of p5 is designed it forces you to some extent to fall back to magic numbers, because wrapping values in constants is cumbersome and doesn't add value.